pkg/didcomm/messaging/service/http/httpmsgsvc.go

```diff
@@ -19,35 +22,117 @@ const (
 	// OverDIDCommSpec is http over DIDComm Spec value
 	OverDIDCommSpec = "https://didcomm.org/http-over-didcomm/1.0/"
 
-	// httpOverDIDComm is http over DIDComm message service name
-	httpOverDIDComm = "http-over-didcomm"
+	// error messages
+	errNameAndHandleMandatory   = "service name and http request handle is mandatory"
+	errFailedToDecodeMsg        = "unable to decode DID comm message: %w"
+	errFailedToDecodeBody       = "unable to decode message body: %w"
+	errFailedToCreateNewRequest = "failed to create http request from incoming message: %w"
 )
 
-// NewHTTPOverDIDComm creates new HTTP over DIDComm message service which serves
-// incoming DIDComm message over HTTP.
-func NewHTTPOverDIDComm() *OverDIDComm {
-	return &OverDIDComm{httpOverDIDComm}
+// RequestHandle handle function for http over did comm message service which gets called by
+// `OverDIDComm` message service to handle incoming request.
+// Args
+// msgID : message ID of incoming message.
+// request: http request derived from incoming DID comm message.
+//
+// Returns
+// error : handle can return error back to service to notify message dispatcher about failures.
+type RequestHandle func(msgID string, request *http.Request) error
+
+// NewDIDCommToHTTPMessageService creates new HTTP over DIDComm message service which serves
+// incoming DIDComm message over HTTP. DIDComm message receiver of [RFC-0351]
+//
+// Reference:
+// https://github.com/hyperledger/aries-rfcs/blob/master/features/0335-http-over-didcomm/README.md
+// https://github.com/hyperledger/aries-rfcs/blob/master/features/0351-purpose-decorator/README.md
+//
+// Args:
+// name: service name, this is mandatory argument.
+//
+// purpose: optional list of purposes to be handled by this http-over-didcomm service [RFC-0351].
+// If not provided only message type will be taken into consideration in acceptance criteria of this message service.
+//
+// httpHandle: handle function to which incoming DIDComm message will be sent after being converted to
+// http request. this is mandatory argument.
+//
+// Returns:
+// OverDIDComm: http over didcomm message service
+// error: arg validation errors
+func NewDIDCommToHTTPMessageService(name string, purpose []string, httpHandle RequestHandle) (*OverDIDComm, error) {
```

Comment:
consider placing `purpose` a variadic arg at the end of the arg list to make it optional

Reply:
fixed